src/test/java/com/powsybl/caseserver/service/AbstractCaseControllerTest.java

```
@@ -724,8 +724,16 @@ private static String getDateSearchTerm(String entsoeFormatDate) {
     @Test
     void invalidFileInCaseDirectoryShouldBeIgnored() throws Exception {
         createStorageDir();
+
+        // add a random file in the storage, not stored in a UUID named directory
         Path filePath = fileSystem.getPath(caseService.getRootDirectory()).resolve("randomFile.txt");
         Files.createFile(filePath);
+
+        // add a case file in a UUID named directory but no metadata in the database
```

I think this test does not work, I've tried with the old code and it passes

---

Yes the file was empty. So an exception was thrown in getCaseInfos(). We want here to check that exceptions in getCaseInfosSafely() are swallowed

---

And let's make it clean. I should adapt this test for S3

---

I moved invalidFileInCaseDirectoryShouldBeIgnored() to FsCaseControllerTest as this test as is is only relevant for FS mode. I'll do a PR to make it relevant for S3.
I created a new test casesWithoutMetadataShouldBeIgnored() to test the code of this PR (a case present in the storage but without metadata in the DB). This one works for both FS and S3. And this test now failed on main branch. See https://github.com/powsybl/powsybl-case-server/pull/68